offload/src/omptarget.cpp

```diff
@@ -1197,6 +1197,82 @@ class PrivateArgumentManagerTy {
   }
 };
 
+/// Try to find redundant mappings associated with a kernel launch,
+/// and provide a masked version of the kernel argument types that
+/// avoid redundant to data transfers between the host and device.
+static std::unique_ptr<int64_t[]>
+maskRedundantTransfers(DeviceTy &Device, int32_t ArgNum, int64_t *ArgTypes,
+                       int64_t *ArgSizes, map_var_info_t *ArgNames,
+                       void **ArgPtrs, void **ArgMappers) {
+  std::unique_ptr<int64_t[]> ArgTypesOverride =
+      std::make_unique<int64_t[]>(ArgNum);
+
+  MappingInfoTy &MappingInfo = Device.getMappingInfo();
+  MappingInfoTy::HDTTMapAccessorTy HDTTMap =
+      MappingInfo.HostDataToTargetMap.getExclusiveAccessor();
+
+  int64_t UnusedArgs = 0;
```

Comment:
This is not a good way to check for a "LastWasUnused". Make it explicit. Also, do we have a test where the object is used but the pointee is not?

Reply:
I'm not checking for "LastWasUnused". I'm checking for "AllAreUnused". Anyway, I made the check more explicit. Let me know if this is OK